src/compiler/program.ts

```diff
@@ -874,12 +874,12 @@ namespace ts {
             return oldProgram.structureIsReused = StructureIsReused.Completely;
         }
 
-        function getEmitHost(writeFileCallback?: WriteFileCallback): EmitHost {
+        function getEmitHost(writeFileCallback?: WriteFileCallback, getCurrentDirectoryCallback?: GetCurrentDirectoryCallback): EmitHost {
             return {
                 getCanonicalFileName,
                 getCommonSourceDirectory: program.getCommonSourceDirectory,
                 getCompilerOptions: program.getCompilerOptions,
-                getCurrentDirectory: () => currentDirectory,
+                getCurrentDirectory: getCurrentDirectoryCallback || (() => currentDirectory),

```

Comment:
`currentDirectory` comes from the `CompilerHost`. Wouldn't this mean that the current directory reported by the compiler host is also incorrect?